### DISCUSSION TITLE
What the comment "it will delete by reclaim" means when meta client removeSession fail?

### DISCUSSION THREAD

**Discussion by iiibui** [vesoft-inc/nebula#3567] [Q&A] [answered]
https://github.com/vesoft-inc/nebula/blob/544c86b9d00593d17963ede999513271c3ba34bb/src/graph/session/GraphSessionManager.cpp#L140-L153

It means meta client will retry in [reclaimExpiredSessions](https://github.com/vesoft-inc/nebula/blob/544c86b9d00593d17963ede999513271c3ba34bb/src/graph/session/GraphSessionManager.cpp#L164-L199)? But it will have no chance because session has been removed from activeSessions_.

It means meta server will do reclaim? But I didn't find any “reclaim” related operations in metad.

**Comment by wey-gu** [comment]
I agree with you that it will be still erased from activeSessions_ anyway including even the case remove session failed. Thus reclaim won't consider this session anymore.

**Reply by yixinglu** [reply]
@wey-gu could you open an issue for this session problem?

**Reply by wey-gu** [reply]
Sure, will create it now.

**Comment by yixinglu** [comment, marked as answer]
@iiibui you are right! I have checked related implementation, I agree with you and think it's a bug. Could you given a PR to fix it? thanks for this feedback.

**Reply by iiibui** [reply]
Sorry I can’t give PR at the moment, because I’m working on other things.
It seems that it is simple to redo the removal on graphd, but I think metad should also do “reclaim”, this is a bit troublesome.

**Reply by wey-gu** [reply]
no worries @iiibui the issue was assigned https://github.com/vesoft-inc/nebula/issues/3612

thanks~~
